flocker/acceptance/test_api.py

```diff
@@ -369,6 +369,57 @@ def can_connect():
         d.addCallback(check_result)
         return d
 
+    def test_create_container_with_environment(self):
+        """
+        Create a container including environment variables on a single-node
+        cluster.
+        """
+        data = {
+            u"name": "my_env_container",
+            u"host": None,
+            u"image": "busybox:latest",
+            u"environment": {u"ACCEPTANCE_ENV_LABEL": 'acceptance test ok'}
+        }
+        waiting_for_cluster = wait_for_cluster(test_case=self, node_count=1)
+
+        def create_container(cluster, data):
+            data[u"host"] = cluster.nodes[0].address
+            return cluster.create_container(data)
+
+        d = waiting_for_cluster.addCallback(create_container, data)
+
+        def check_result(result):
+            response = result[1]
```

Comment:
I think it would be slightly clearer to unpack the cluster, response in the callback signature eg

```
def check_result((cluster, response)):
    ...
```

And that ensures that you'd get an error if the result ever contains an unexpected number of parts.